src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2433,61 +2435,16 @@ public String restrictSelectedFiles(boolean restricted) throws CommandException{
         return  returnToDraftVersion();
     }
 
-    public void restrictFiles(boolean restricted) throws CommandException {
-   
-        //if (previouslyRestrictedFiles == null) {
-        // we don't need to buther with this "previously restricted" business 
-        // when in Create mode... because all the files are new, so none could 
-        // have been restricted previously;
-        // (well, it looks like the code below should never be called in the 
-        // CREATE mode in the first place... the edit files fragment uses
-        // its own restrictFiles() method there; also, the fmd.getDataFile().equals(fmw.getDataFile()))
-        // line is not going to work on a new file... so be mindful of all this
-        // when the code between the 2 beans is merged in 4.3.
-        if (editMode != EditMode.CREATE) {
-            previouslyRestrictedFiles = new ArrayList<>();
-            for (FileMetadata fmd : workingVersion.getFileMetadatas()) {
-                if (fmd.isRestricted()) {
-                    previouslyRestrictedFiles.add(fmd);
-                }
+    private void restrictFiles(boolean restricted) throws CommandException {
+        Command<Void> cmd;
+        previouslyRestrictedFiles = new ArrayList<>();
+        for (FileMetadata fmd : this.getSelectedFiles()) {
+            if(fmd.isRestricted()) {
+                previouslyRestrictedFiles.add(fmd);
             }
-            
-            Command cmd;
-            String fileNames = null;
-            for (FileMetadata fmw : workingVersion.getFileMetadatas()) {
-                for (FileMetadata fmd : this.getSelectedFiles()) {
-                    if (restricted && !fmw.isRestricted()) {
-                    // collect the names of the newly-restrticted files, 
```

Comment:
because this is called from restrictSelectedFiles above that calls save() which already sets a success message, this list of filenames is never used/never shown because the attempt to add a flash message below fails (an error in the log).